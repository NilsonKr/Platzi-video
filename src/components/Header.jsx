import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import gravatar from '../utils/gravatar';

import logo from '@assets/logo-platzi-video-BW2.png';
import userIcon from '@assets/user-icon.png';
import '@styles/components/Header.scss';

const Header = props => {
	console.log(props);
	const hasUser = Object.keys(props.user).length > 0;

	return (
		<header className='header'>
			<Link to='/'>
				<img className='header__img' src={logo} alt='Platzi Video' />
			</Link>
			<div className='header__menu'>
				<div className='header__menu--profile'>
					<img src={hasUser ? gravatar(props.user.email) : userIcon} alt='User' />
					<p>Profile</p>
				</div>
				<ul>
					<li>
						<a href='/'>Account</a>
					</li>
					<li>
						<Link to='/login'>Log In</Link>
					</li>
				</ul>
			</div>
		</header>
	);
};

const mapStateToProps = state => ({ user: state.user });

export default connect(mapStateToProps)(Header);
