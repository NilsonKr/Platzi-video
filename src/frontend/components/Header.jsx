import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { handleLogOut } from '../actions/actions';
import gravatar from '../utils/gravatar';
import classNames from 'classnames';

import '../styles/components/Header.scss';

const Header = props => {
	const hasUser = Object.keys(props.user).length > 0;

	const classHeader = classNames('header', [`header--${props.bg}`]);

	return (
		<>
			<header className={classHeader}>
				<Link to='/'>
					<img className='header__img' src='assets/logo-platzi-video-BW2.png' alt='Platzi Video' />
				</Link>
				<div className='header__menu'>
					<div className='header__menu--profile'>
						<img src={hasUser ? gravatar(props.user.email) : 'assets/user-icon.png'} alt='User' />
						<p>{props.user.name || 'Profile'}</p>
					</div>
					<ul>
						{hasUser ? (
							<li>
								<Link to='/login'>Account</Link>
							</li>
						) : null}
						{hasUser ? (
							<li onClick={props.handleLogOut}>
								<a>Log Out</a>
							</li>
						) : (
							<li>
								<Link to='/login'>Log In</Link>
							</li>
						)}
					</ul>
				</div>
			</header>
		</>
	);
};

const mapStateToProps = state => ({ user: state.user });
const mapDispatchToProps = {
	handleLogOut,
};

export default connect(mapStateToProps, mapDispatchToProps)(Header);
