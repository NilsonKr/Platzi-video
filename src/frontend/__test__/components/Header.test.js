import React from 'react';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import Header from '../../components/Header';
import ProviderMock from '../../__mocks__/providerMock';

describe('<Header />', () => {
	const headerMounted = mount(
		<ProviderMock>
			<Header />
		</ProviderMock>
	);

	test('Header Logo Image', () => {
		expect(headerMounted.find('.header__img')).toHaveLength(1);
	});

	test('Header Snapshot', () => {
		const headerSnap = create(
			<ProviderMock>
				<Header />
			</ProviderMock>
		);

		expect(headerSnap.toJSON()).toMatchSnapshot();
	});
});
