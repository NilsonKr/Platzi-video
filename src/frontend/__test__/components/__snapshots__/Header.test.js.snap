// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Header Snapshot 1`] = `
<header
  className="header header--undefined"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Platzi Video"
      className="header__img"
      src="assets/logo-platzi-video-BW2.png"
    />
  </a>
  <div
    className="header__menu"
  >
    <div
      className="header__menu--profile"
    >
      <img
        alt="User"
        src="assets/user-icon.png"
      />
      <p>
        Profile
      </p>
    </div>
    <ul>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Log In
        </a>
      </li>
    </ul>
  </div>
</header>
`;
